pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -1194,9 +1195,25 @@ def _process_response(  # noqa: C901
             elif isinstance(item, responses.ResponseComputerToolCall):  # pragma: no cover
                 # Pydantic AI doesn't yet support the ComputerUse built-in tool
                 pass
-            elif isinstance(item, responses.ResponseCustomToolCall):  # pragma: no cover
-                # Support is being implemented in https://github.com/pydantic/pydantic-ai/pull/2572
-                pass
+            elif isinstance(item, responses.ResponseCustomToolCall):
+                # Handle custom tool calls (freeform function calling)
+                if item.name not in model_request_parameters.tool_defs:
```

Comment:
Let's make it clear this is for the scenario where the model calls a non-existent tool